### ISSUE TITLE
Add @DocumentId annotation to firestore for multiplatform

### ISSUE THREAD

**Issue by mr-kew** [GitLiveApp/firebase-kotlin-sdk#445]
| Library  | Class            | Member               | Platforms        |
|:---------|:-----------------|:---------------------|:-----------------|
| firestore | annotations | DocumentId | Android, iOS |

Add a way to get documentId as a property inside `@Serializable` class.

It is done using [annotation in native kotlin library](https://firebase.google.com/docs/reference/kotlin/com/google/firebase/firestore/DocumentId) and using [property wrapper in native swift library](https://firebase.google.com/docs/firestore/solutions/swift-codable-data-mapping).

**Comment by adityatri** [comment]
Would like to see this as well. Currently I'm not able to use `@DocumentId` annotation for Android and iOS.

**Comment by zvona031** [comment]
Any work on this?

**Comment by mr-kew** [comment]
@zvona031 Not from my side, I just did a workaround in my code. 

I used `@Transient` property, that is manually set when deserializing the data.

**Comment by AdamMc331** [comment]
@mr-kew Do you happen to have a sample of this? My work around was to add a separate `id` field onto the document itself, which is fine, but curious what the Transient setup looks like.

**Comment by mr-kew** [comment]
@AdamMc331 Yeah, the idea is pretty simple. I use some constant string (e.g. `auto_id`) as a placeholder for the `DocumentId` while the data gets serialized and then create a copy of the data class with the correct `DocumentId`.

```
@Serializable
data class SomeDocument(
    @Transient val id: String = "auto_id",
    ...
)

suspend fun getSomeDocument(id: String): SomeDocument {
    val document = Firebase.firestore
        .collection("someDocuments")
        .document(id)
        .get()
    return document
        .data<SomeDocument>()
        .copy(id = document.id)
}
```

I mean... it's not perfect, but adding separate id field into the document seems pretty... radical

**Comment by AdamMc331** [comment]
@mr-kew Worked like a charm, thanks for your help!
